mmv1/api/resource.go

```diff
@@ -730,14 +730,90 @@ func (r Resource) GetIdentity() []*Type {
 	})
 }
 
-func (r *Resource) AddLabelsRelatedFields(props []*Type, parent *Type) []*Type {
+func buildWriteOnlyField(name string, originalField *Type, originalFieldLineage string) *Type {
+	description := fmt.Sprintf("%s Note: This property is write-only and will not be read from the API. For more info see [updating write-only attributes](/docs/providers/google/guides/using_write_only_attributes.html#updating-write-only-attributes)", google.Underscore(originalField.Description))
+	fieldPathOriginalField := originalFieldLineage
+	fieldPathCurrentField := strings.ReplaceAll(originalFieldLineage, google.Underscore(originalField.Name), google.Underscore(name))
+
+	apiName := originalField.ApiName
+	if apiName == "" {
+		apiName = originalField.Name
+	}
+
+	options := []func(*Type){
+		propertyWithType("String"),
+		propertyWithRequired(false),
+		propertyWithDescription(description),
+		propertyWithWriteOnly(true),
+		propertyWithApiName(apiName),
+		propertyWithIgnoreRead(true),
+	}
+
+	if originalField.Required {
+		exactlyOneOf := append(originalField.ExactlyOneOf, fieldPathOriginalField, fieldPathCurrentField)
```

Comment:
If the original field already has an exactlyoneof, it should already be included there, so we only need to add the current field.

```suggestion
		exactlyOneOf := append(originalField.ExactlyOneOf, fieldPathCurrentField)
```

Reply:
hmm, no, I was getting confused because of the `append` - you're right, if it's just `required` with no existing `exactly_one_of` constraint, we need to make sure both fields are present. Thanks!

I'm not sure whether it will cause problems to extend an existing exactly_one_of like this or not... but also, in that case `required` might not be set at all, so this branch wouldn't even activate.

Let's just leave this as-is for now.